packages/jazz-tools/src/react-native/media/image.tsx

```diff
@@ -0,0 +1,107 @@
+import { FileStream, ImageDefinition } from "jazz-tools";
+import { highestResAvailable } from "jazz-tools/media";
+import { useCoState } from "jazz-tools/react-native-core";
+import { forwardRef, useEffect, useMemo, useState } from "react";
+import { Image as RNImage, ImageProps as RNImageProps } from "react-native";
+
+export type ImageProps = Omit<RNImageProps, "width" | "height" | "source"> & {
+  imageId: string;
+  width?: number | "original";
+  height?: number | "original";
+};
+
+const Image = forwardRef<RNImage, ImageProps>(function Image(
+  { imageId, width, height, ...props },
+  ref,
+) {
+  const image = useCoState(ImageDefinition, imageId);
+  const [src, setSrc] = useState<string | undefined>(image?.placeholderDataURL);
+
+  const dimensions: { width: number | undefined; height: number | undefined } =
+    useMemo(() => {
+      const originalWidth = image?.originalSize?.[0];
+      const originalHeight = image?.originalSize?.[1];
+
+      // Both width and height are "original"
+      if (width === "original" && height === "original") {
+        return { width: originalWidth, height: originalHeight };
+      }
+
+      // Width is "original", height is a number
+      if (width === "original" && typeof height === "number") {
+        if (originalWidth && originalHeight) {
+          return {
+            width: Math.round((height * originalWidth) / originalHeight),
+            height,
+          };
+        }
+        return { width: undefined, height };
+      }
+
+      // Height is "original", width is a number
+      if (height === "original" && typeof width === "number") {
+        if (originalWidth && originalHeight) {
+          return {
+            width,
+            height: Math.round((width * originalHeight) / originalWidth),
+          };
+        }
+        return { width, height: undefined };
+      }
+
+      // In all other cases, use the property value:
+      return {
+        width: width === "original" ? originalWidth : width,
+        height: height === "original" ? originalHeight : height,
+      };
+    }, [image?.originalSize, width, height]);
+
+  useEffect(() => {
+    if (!image) return;
+
+    let lastBestImage: FileStream | string | undefined =
+      image.placeholderDataURL;
+
+    const unsub = image.subscribe({}, (update) => {
```

Comment:
I think it would be better to not have the image as dependency and fully subscribe to the image internally.

```suggestion
    const unsub = ImageDefinition.subscribe(imageId, {}, (update) => {
```

This way the dependencies are more stable and the component re-renderings more predictable.